tenant_schemas/middleware.py

```diff
@@ -75,22 +73,28 @@ class SuspiciousTenantMiddleware(TenantMiddleware):
     TENANT_NOT_FOUND_EXCEPTION = DisallowedHost
 
 
-class DefaultSchemaTenantMiddleware(TenantMiddleware):
+class DefaultTenantMiddleware(SuspiciousTenantMiddleware):
```

Comment:
If the configured schema doesn't exist, a 404 will go on silently. A 400 on the other hand is an error condition that will be logged, and often result in mail to ADMINS.

It's a defensive move, I can imagine a case where if a non-public schema is chosen, there is a possibility that it might be deleted; site admins should probably hear about that.